src/Razor/test/Microsoft.AspNetCore.Razor.LanguageServer.Test/CodeActions/Razor/ExtractToCodeBehindCodeActionProviderTest.cs

```diff
@@ -337,10 +347,11 @@ public async Task Handle_NullRelativePath_ReturnsNull()
             """;
         TestFileMarkupParser.GetPosition(contents, out contents, out var cursorPosition);
 
-        var request = new CodeActionParams()
+        var request = new VSCodeActionParams()
         {
-            TextDocument = new TextDocumentIdentifier { Uri = new Uri(documentPath) },
+            TextDocument = new VSTextDocumentIdentifier { Uri = new Uri(documentPath) },
```

Comment:
For future reference, is there any reason why we should ever do `TextDocumentIdentifier` instead of `VSTextDocumentIdentifier`? It seems like if someone doesn't want the extra VS stuff (and IIRC it's all additive) they'll just drop it when they serialize right?

Reply:
Personally, I think this whole PR shouldn't exist and we should never use `VS*` types. I think its a failing of the platform that we have to.

Reply:
Internestingly Web Tools has a `VSCodeActionParams` of their own, presumably to acheive the same thing as this PR, but they use `TextDocumentIdentifier`. I'm just more lazy and did a direct copy/paste.